ls8/cpu.c

```diff
@@ -103,16 +104,8 @@ void cpu_ram_write(struct cpu *cpu, unsigned char address, unsigned char value)
  */
 
 int num_operands_needed(unsigned char IR) {
-  switch (IR) {
-    case LDI:
-      return 2;
-
-    case PRN: 
-      return 1;
-
-    default:
-      return 0;
-  }
+  unsigned char AA = IR>>6;
```

Comment:
it's not critical, but why all the change of types?
something like `unsigned char num_operands = IR >> 6;` is fine